/*
 * This file was automatically generated by EvoSuite
 * Mon Aug 19 20:47:48 GMT 2019
 */

package rogeriogentil.service;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import javax.security.auth.login.FailedLoginException;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;
import rogeriogentil.service.AuthenticationImpl;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class AuthenticationImpl_ESTest extends AuthenticationImpl_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      AuthenticationImpl authenticationImpl0 = new AuthenticationImpl();
      // Undeclared exception!
      try { 
        authenticationImpl0.authenticate("", (String) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("rogeriogentil.repository.UserRepository", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      AuthenticationImpl authenticationImpl0 = new AuthenticationImpl();
      try { 
        authenticationImpl0.authenticate("", "");
        fail("Expecting exception: FailedLoginException");
      
      } catch(FailedLoginException e) {
         //
         // Invalid Credentials.
         //
         verifyException("rogeriogentil.service.AuthenticationImpl", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      AuthenticationImpl authenticationImpl0 = new AuthenticationImpl();
      authenticationImpl0.authenticate("", "123456");
  }
}
